lua/lsp-selection-range/init.lua

```diff
@@ -73,9 +89,7 @@ function M.update_capabilities(capabilities)
   vim.validate({ capabilities = { capabilities, 'table', true } })
   capabilities = capabilities or vim.lsp.protocol.make_client_capabilities()
 
-  capabilities.textDocument = vim.tbl_extend('force', capabilities.textDocument or {}, {
-    selectionRange = { dynamicRegistration = false },
-  })
+  capabilities = vim.tbl_deep_extend('force', capabilities or {}, M.default_capabilities(nil))
 
   return capabilities
```

Comment:
Given the previous comment I would think of something like:
```lua
  vim.validate({ capabilities = { capabilities, 'table', true } })
  if nil == capabilities then
    -- throw a deprecation => nil won't be handled anymore (it was a mistake)
  end
  capabilities = capabilities or {}

  capabilities.textDocument = vim.tbl_deep_extend('keep', capabilities.textDocument or {}, {
    selectionRange = { dynamicRegistration = false },
  })

  return capabilities
```

As explained I think the user should always provide default capabilities, hence the deprecation.
I also think I made a mistake by forcing the merge. Instead we should keep the values provided by the user. Even though I'm not sure it made much sense to have another client which will handle the dynamic registration I don't think it should be the plugin's responsibility to change things it does not requires in order to work.

Which means a new test needs to be added to guarantee that we won't change the capabilities if `.textDocument.selectionRange.dynamicRegistration` is already provided in the capabilities to update.

Reply:
Agreed (as long as `dynamicRegistration = true` doesn't break any of this plugin's functionality).

Reply:
I would suggest throwing a deprecation, but still defaulting to `vim.lsp.protocol.make_client_capabilities()` for backward compatibility for now.